app/src/main/java/com/mailerdaemon/app/campusMap/MapsActivity.kt

```diff
@@ -0,0 +1,272 @@
+package com.mailerdaemon.app.campusMap
+
+import android.Manifest
+import android.app.Activity
+import android.content.DialogInterface
+import android.content.pm.PackageManager
+import android.location.Location
+import android.os.Build
+import android.os.Bundle
+import android.os.Looper
+import android.view.MenuItem
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.google.android.gms.location.*
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.*
+import com.mailerdaemon.app.R
+import com.mailerdaemon.app.campusMap.MapsActivity
+import java.lang.ref.WeakReference
+import java.util.*
+
+class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
+    var mLocationRequest: LocationRequest? = null
+    var mLastLocation: Location? = null
+    var mCurrLocationMarker: Marker? = null
+    var mFusedLocationClient: FusedLocationProviderClient? = null
+    var mLocationCallback: LocationCallback? = null
+    var ref = WeakReference<Activity>(this)
+    private var mMap: GoogleMap? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_maps)
+        Objects.requireNonNull(supportActionBar)?.setDisplayShowHomeEnabled(true)
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        val mapFragment = (supportFragmentManager
+                .findFragmentById(R.id.map) as SupportMapFragment?)!!
+        mapFragment.getMapAsync(this)
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+        supportActionBar!!.title = "Campus Map"
```

Comment:
Use 'let{}' to perform multiple operations on same object.

Reply:
Strings which are to be shown in ui should be present in String resources instead of hard coded